src/netif.c

```diff
@@ -276,60 +307,100 @@ static void encode_kv_operstate(struct netif *nb, int operstate)
     ei_encode_atom(nb->resp, &nb->resp_index, operstate_atom);
 }
 
-static int netif_build_ifinfo(const struct nlmsghdr *nlh, void *data)
+static int ethtool_gset_ioctl(struct netif *nb, const char *ifname, struct netif_link_settings *ls)
 {
-    struct netif *nb = (struct netif *) data;
-    struct nlattr *tb[IFLA_MAX + 1];
-    memset(tb, 0, sizeof(tb));
-    struct ifinfomsg *ifm = mnl_nlmsg_get_payload(nlh);
+    struct ethtool_cmd ecmd = {0, };
+    struct ifreq ifr = {0, };
 
-    if (mnl_attr_parse(nlh, sizeof(*ifm), collect_if_attrs, tb) != MNL_CB_OK) {
-        debugf("Error from mnl_attr_parse");
-        return MNL_CB_ERROR;
+    ecmd.cmd = ETHTOOL_GSET;
+
+    strncpy(ifr.ifr_name, ifname, IFNAMSIZ-1);
+    ifr.ifr_data = (void *) &ecmd;
+
+    if (ioctl(nb->inet_fd, SIOCETHTOOL, &ifr) < 0) {
+        error("ioctl(0x%04x) failed for getting '%s': %s for %s", SIOCETHTOOL, "ETHTOOL_GSET", strerror(errno), ifname);
+        nb->last_error = errno;
+        return -1;
     }
 
-    int count = 8; /* Number of fields that we always encode */
-    int i;
+    ls->speed  = (ecmd.speed_hi << 16) | ecmd.speed;;
```

Comment:
any reason for the extra ;; ?

Reply:
To be absolutely certain the sentence is terminated ;) Joking - corrected...